OpenRA.Mods.Common/Traits/Air/TargetableAircraft.cs

```diff
@@ -31,13 +31,9 @@ public TargetableAircraft(Actor self, TargetableAircraftInfo info)
 			this.self = self;
 		}
 
-		public override string[] TargetTypes
+		protected override string[] GetTargetableTypes()
```

Comment:
This plumbing change makes the already complicated targeting plumbing a step worse, IMO.  Keep it as a property without any inner methods, and do the disabled check in these two extra places.